### PR TITLE
🛠️ Rework tests

### PR DESCRIPTION
**Pull request by JuaniRios**
## What?
- Fix pallet and integration tests
- Improve instantiator flow
- Merged community and remainder test files into one

## To reviewers:
- You will have to go review all the tests since a lot of the changed significantly, and the diff is kinda broken, sry

### CODE REVIEW

**Comment by JuaniRios** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/365?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#365** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/365?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#364** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/364?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#363** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/363?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#362** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/362?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#361** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/361?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#358** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/358?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JuaniRios and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: pallets/funding/src/tests/runtime_api.rs [hunk 1]
```diff
@@ -646,15 +648,16 @@ fn usd_target_percent_reached() {
 fn projects_by_did() {
 	let mut inst = MockInstantiator::new(Some(RefCell::new(new_test_ext())));
 	let did_user = generate_did_from_account(420);
+	dbg!(&did_user);
```
Comment (lrazovic) [inline pallets/funding/src/tests/runtime_api.rs:651]:
```suggestion
```

File: pallets/funding/src/tests/runtime_api.rs [hunk 2]
```diff
@@ -683,11 +686,13 @@ fn projects_by_did() {
...
 		default_bids(),
-		default_community_buys(),
-		default_remainder_buys(),
+		default_community_contributions(),
+		default_remainder_contributions(),
 	);
 
 	inst.execute(|| {
+		let projects = ProjectsDetails::<TestRuntime>::iter().collect_vec();
+		dbg!(projects);
```
Comment (lrazovic) [inline pallets/funding/src/tests/runtime_api.rs:695]:
```suggestion
```

**Review by lrazovic — APPROVED** [review]
Few nits here and there, but overall looks good.

File: pallets/funding/src/tests/5_funding_end.rs [hunk 3]
```diff
@@ -7,40 +7,46 @@ mod round_flow {
...
+						inst.get_project_details(project_id).status,
+						ProjectStatus::SettlementStarted(FundingOutcome::Success)
+					);
+					assert!(matches!(
+						inst.get_project_details(project_id).evaluation_round_info.evaluators_outcome,
+						Some(EvaluatorsOutcome::Rewarded(..))
+					));
+				}
+			};
+			for i in 10..=32u8 {
```
[the remark this answers is not in the document]
Comment (JuaniRios) [reply pallets/funding/src/tests/5_funding_end.rs:33]: Was worried of bad setup behavior with low percentage. 0 fails, but 1 is ok, changed it from 1..32

File: pallets/funding/src/tests/6_settlement.rs
Comment (vstam1) [inline pallets/funding/src/tests/6_settlement.rs:1]: Please remove all dbg statements

**Review by vstam1 — APPROVED** [review]
Approved with a couple of comments about removing dbg statements

**Pull request merged**